### PR TITLE
Build/yarn and upgrades

### PR DESCRIPTION
**Pull request by Gijsdeman**
This pull requests adds the following:

- Switch from `npm` to `yarn`
- Updates all dependencies, and fixes all build errors (and deprecation warnings)
- Switch to latest long term support of node (22.11)

### CODE REVIEW

**Comment by Gijsdeman** [comment]
Will have to upgrade `aurora-core` to node 22 to properly work with the updated workflows.

File: tsconfig.json.bac [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (Yoronex) [inline tsconfig.json.bac:1]: Is this file doing anything? I have never seen a `.bac` file...

**Review by Yoronex — APPROVED** [review]

**Comment by Yoronex** [comment]
Oh I forgot, also change the node version in the prerequisites in `README.md`!

File: tsconfig.json.bac (hunk 1 above)
Comment (Gijsdeman) [reply tsconfig.json.bac:1]: oops, accidental backup file to check some settings. Will remove it.

**Comment by Gijsdeman** [comment]
Lint fails because of split `tsconfig`. However, this is fixed with the shared config PR. I do want to keep those two separated for now; so it should be fine to ignore this error right now.

**Pull request merged**